ndonnx/_core/_impl.py

```diff
@@ -1029,6 +1037,34 @@ def _via_i64_f64(
     return _variadic_op(arrays, fn, via_dtype, cast_return)
 
 
+def _via_i64_f32(
```

Comment:
Can you share some of the logic between this and `_via_i64_f64`. Specifically, I think providing the types as input and making sure you don't downcast (you can check `ndx.iinfo(dtype).bits`/`ndx.finfo(dtype).bits`) is sufficiently general. It would then make it easier to do more upcasting permutations as and when we find the need.

Reply:
Yeah makes sense, I'll do that.